switchapi/inc/switchapi/switch_hostif.h

```diff
@@ -199,6 +198,67 @@ switch_api_hostif_delete(switch_device_t device, switch_handle_t hostif_handle);
 switch_handle_t
 switch_api_cpu_nhop_get(switch_hostif_reason_code_t rcode);
 
+typedef enum switch_packet_vlan_action {
+    SWITCH_PACKET_VLAN_NONE        = 0x0,
+    SWITCH_PACKET_VLAN_ADD         = 0x1,
+    SWITCH_PACKET_VLAN_REMOVE      = 0x2,
+    SWITCH_PACKET_VLAN_SWAP        = 0x3
+} switch_packet_vlan_action_t;
+
+typedef struct switch_packet_rx_key_ {
+    bool port_valid;
+    switch_handle_t port_handle;                           /**< port handle */
+    bool port_lag_valid;
+    switch_handle_t port_lag_handle;                       /**< port or lag handle */
+    bool handle_valid;
+    switch_handle_t handle;                                /**< bd or interface handle */
+    bool reason_code_valid;
+    switch_hostif_reason_code_t reason_code;               /**< reason code */
+    uint32_t priority;
+} switch_packet_rx_key_t;
+
+typedef struct switch_packet_rx_action_ {
+    switch_handle_t hostif_handle;
+    switch_vlan_t vlan_id;
+    switch_packet_vlan_action_t vlan_action;
+} switch_packet_rx_action_t;
+
+typedef struct switch_packet_tx_key_ {
+    bool handle_valid;
+    switch_handle_t hostif_handle;
+    bool vlan_valid;
+    switch_vlan_t vlan_id;
+    uint32_t priority;
+} switch_packet_tx_key_t;
+
+typedef struct switch_packet_tx_action_ {
+    switch_handle_t handle;                                /**< bd or interface handle */
+    bool tx_bypass;                                        /**< tx bypass */
```

Comment:
Why not take the bypass_bitmap here instead of a bool? (Future proof?)
Or did we want to call it tx_hw_acceleration ?